plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -227,4 +237,52 @@ private static String quoteIdentifier(String identifier)
     {
         return format("\"%s\"", identifier.replaceAll("\"", "\"\""));
     }
+
+    public static Optional<String> getQueryOptionsString(String options)
+    {
+        if (isNullOrEmpty(options)) {
+            return Optional.empty();
+        }
+
+        Map<String, String> queryOptionsMap = parseQueryOptions(options);
+        return getQueryOptions(queryOptionsMap);
+    }
+
+    public static Optional<String> getQueryOptions(Map<String, String> queryOptionsMap)
+    {
+        if (queryOptionsMap.isEmpty()) {
+            return Optional.empty();
+        }
+        String options = queryOptionsMap.entrySet().stream()
+                .map(e -> "SET " + e.getKey() + " = " + e.getValue())
+                .collect(Collectors.joining(";\n"));
+        if (!options.endsWith(";")) {
+            options += ";\n";
+        }
+        return Optional.of(options);
+    }
+
+    public static Map<String, String> parseQueryOptions(String options)
+    {
+        if (isNullOrEmpty(options)) {
```

Comment:
I don't see why the parsing needs to be done so late - do it in `PinotConfig` as I've already suggested and return a `Map<String, String>` for the query options.

Reply:
I dont see a reason to return Map. All the callers need is a string to be prepended to the SQL statement. Map is not relevant. Map is not useful at all.

Reply:
@findinpath this is not something I would want to do.

Reply:
I guess I misunderstood your comment. Upon reading it again addressed that. @findinpath please review last commit.